src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -90,6 +92,20 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  dout(20) << __func__ << ": " << "sending NBD_DISCONNECT" << dendl;
+  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
+    derr << "rbd-nbd: disconnect failed" << cpp_error(errno) << std::endl;
+  } else {
+    dout(20) << __func__ << ": " << "disconnected" << dendl;
+  }
```

Comment:
@yangdongsheng They are different, because in do_unmap case, this is the user who specifies the device name, and the most probable error in this case is that the user specified a wrong device or it had already been closed. In the signal handler if ioctl fails it is definitely not due to the user input, and is the most probably a bug manifestation either in the rbd-nbd or the kernel.